alf/environments/alf_wrappers.py

```diff
@@ -933,3 +933,34 @@ def _step(self, action):
         # action: [B, action_dim] or [action_dim]
         time_step = self._env.step(action)
         return time_step._replace(prev_action=prev_action)
+
+
+class AtariTerminalOnLifeLossWrapper(AlfEnvironmentBaseWrapper):
+    """Wrapper to change discount to 0 upon life loss for Atari.
+
+    This can potentially make it easier for the learning agent to recognize the
+    signficance of losing a life.
+
+    Some papers report the results with this enabled (e.g. arXiv:2111.00210)
+    """
+
+    def __init__(self, env):
+        """
+        Args:
+            env: ALF env to be wrapped
+            actions_num: number of values to discretize each action dim into
+        """
+        super().__init__(env)
+        self._prev_lives = torch.zeros(env.batch_size, dtype=torch.long)
+
+    def _reset(self):
+        time_step = self._env.reset()
+        self._prev_lives = time_step.env_info['ale.lives']
+        return time_step
+
+    def _step(self, action):
+        time_step = self._env.step(action)
+        lives = time_step.env_info['ale.lives']
```

Comment:
By default, 'ale.lives' might not exist in the env info?

Reply:
gym.envs.atari.atari_env always output "ale.lives".